### PR TITLE
Fix Getting Dimmer Max Value

### PR DESCRIPTION
**Pull request by kecramer**
Dimmer max is a single 16 bit value, not an 8 bit.

### CODE REVIEW

**Comment by kecramer** [comment]
This is failing `make check` but so is master, and I believe it unrelated to this code.

File: common/rdm/RDMAPI.cpp [hunk 1]
```diff
@@ -4413,7 +4413,8 @@ void RDMAPI::_HandleGetDimmerMaximumLevel(
     if (data.size() != DATA_SIZE) {
       SetIncorrectPDL(&response_status, data.size(), DATA_SIZE);
     } else {
-      maximum_level = data[0];
+      const uint16_t *ptr = reinterpret_cast<const uint16_t*>(data.data());
```
Comment (peternewman) [inline common/rdm/RDMAPI.cpp]: Good catch, sorry we both missed it first time around. You can simply this slightly, if you look at how we do _HandleGetDMXAddress you don't need the ptr variable.

**Comment by kecramer** [comment]
```common/utils/DmxBufferTest.cpp:197:10: error: explicitly assigning value of
      variable of type 'ola::DmxBuffer' to itself
      [-Werror,-Wself-assign-overloaded]
  buffer = buffer;
  ~~~~~~ ^ ~~~~~~
1 error generated.```

Is the error when I am up to date with origin/master (and for this branch as well)

File: common/rdm/RDMAPI.cpp (hunk 1 above)
Comment (kecramer) [reply common/rdm/RDMAPI.cpp]: Sounds good, I will tweak that today.

**Comment by peternewman** [comment]
Thanks @kecramer . Sorry I missed it before, but could you switch SetDMXAddress to use the new GenericSetU16 you've very sensibly added.

**Comment by kecramer** [comment]
Ah good point, definitely missed that one. Let me know if you have any other comments.

**Comment by peternewman** [comment]
> Ah good point, definitely missed that one. Let me know if you have any other comments.

Well in fairness I didn't initially suggest even making a generic set as I'd missed the fact the brightness would have a set too.

**Pull request merged**